### DISCUSSION TITLE
Feature: Branch views EDIT: Feature: themes and modules

### DISCUSSION THREAD

**Discussion by SallyMcGrath** [CodeYourFuture/curriculum#394] [Ideas]
**Which module(s) and week(s) does this change affect?**
Module(s): all
Week(s): all

**What is the work that needs to be done?**

I'm considering this only

If we were to create a close fork for MigraCode, what might we want to do? I think we want their content to be available to CYF as well, right? So a common pool of blocks? Should we move blocks to an independent repo: curriculum blocks and refactor this as a platform only repo? Will github actually just block us if we create everything via their api...

Should we create a Hugo theme that swaps out all the CodeYourFuture org links to MigraCodes but expects their modules to be forks of ours?

Should we have a spanish language switcher? That seems pretty old school. I feel like a fork switcher might be better. 

Should we create an independent workshops section anyway, and just pull in all workshops and tag em so mentors can get a view more easily?
**Why is this work important to do?**

If we can understand how to merge with MigraCode, we will be closer to creating a truly useful curriculum & platform.

**Additional context**

**Who might need to know about this change?**

**Comment by SallyMcGrath** [comment]
1. Sally and Daniel Register the main curriculum as a module or series of modules (Hugo modules)
2. Sally Create a parent theme that imports those modules
3. Sally Create two themes as siblings of this parent: MigraCode and CodeYourFuture 
4. Sally Skin your website for you ✨ 
5. CYF Handle the content - what is shared and what is CYF
6. CYF remove pointless cyf branding in text or otherwise
7.  CYF decide what to do about the others that are not pointless
8. Sally Replace backlog with (generically) links to GitHub Classroom. MigraCode to go through and put in specific links
9. MigraCode consider further integration with GHC
10. MigraCode audit and take on adaptation

**Comment by illicitonion** [comment]
Content/topics from the existing MigraCode curriculum which we'd like to migrate over to the curriculum:

- [ ] * [`localStorage` and `sessionStorage`](https://github.com/Migracode-Barcelona/js2-workshop#web-storage-api)
- [ ] * [ES6 modules](https://github.com/Migracode-Barcelona/js3-workshop#es-6-modules) - we probably want to simplify this
- [ ] * [Tour of UI surfaces in VS Code](https://syllabus.migracode.org/courses/introduction-3/course-content/introduction-2/dev_tools#development-tools)
- [ ] * [Git repo creation workshop](https://syllabus.migracode.org/courses/introduction-3/course-content/introduction-2/git-and-github-practice-session)
- [ ] * [Async tests in jest](https://github.com/Migracode-Barcelona/js3-workshop#testing-async-code)
- [ ] * [Middlewares in NodeJS](https://syllabus.migracode.org/courses/introduction-3/course-content/nodejs/week-2#what-does-a-middleware-function-look-like)
- [ ] * [Express Auth + JWT workshop](https://syllabus.migracode.org/courses/introduction-3/course-content/nodejs/week-3) (maybe just as a workshop not folded into the curriculum for now)
- [ ] * Links and references in ~every page

**Reply by illicitonion** [reply]
I've filed issues for each of these.

**Comment by SallyMcGrath** [comment]
CYF and MigraCode define core contributor/ maintainer and register a list and give them powers.

**Comment by SallyMcGrath** [comment]
https://cloudcannon.com/blog/going-wild-with-hugo-modules/
